src/Cooked/Skeleton.hs

```diff
@@ -902,6 +927,13 @@ txSkelOutReferenceScripts =
       )
     . txSkelOuts
 
+-- | All utxos present in the skeleton
+txSkelKnownTxOutRefs :: TxSkel -> [Api.TxOutRef]
+txSkelKnownTxOutRefs TxSkel {..} =
+  Map.keys txSkelIns
+    <> mapMaybe txSkelReferenceScript (Map.elems txSkelIns)
+    <> Set.toList txSkelInsReference
+
```

Comment:
The outputs of the skeleton do not contain any `TxOutRef` as outputs only because TxOutRef once the transaction that produces them has been validated. So in fact the name of this function makes sense and the function indeed collects all `TxOutRef`s known by the transaction skeleton. 

About homogenizing the names: This is already pretty well homogenized, as some helpers work on TxSkelOut and some others work on TxSkel, which their name reflect, but I will make another pass to ensure everything is properly homogeneous.

Reply:
Here is the updated comment:
```
-- | All `TxOutRefs` known by a given transaction skeleton. This includes
-- TxOutRef`s used as inputs of the skeleton and `TxOutRef`s used as reference
-- inputs of the skeleton.  This does not include additional possible
-- `TxOutRef`s used for balancing and additional `TxOutRef`s used as collateral
-- inputs, as they are not part of the skeleton.
```

Reply:
I homogenized the wrong names:
- `txSkelOutputData` to `txSkelDataInOutputs`
- `txSkelOutValidators` to `txSkelValidatorsInOutputs`
- `txSkelOutReferenceScripts` to `toSkelReferenceScripts`
- `txSkelOutputsValue` to `txSkelValueInOutputs`